scipy/stats/stats.py

```diff
@@ -5586,3 +5589,89 @@ def rankdata(a, method='average'):
 
     # average method
     return .5 * (count[dense] + count[dense - 1] + 1)
+
+
+def rank_distance(x, y, weights=None, method='spearman'):
+    """
+    Distance measure between rankings.
+
+    Parameters
+    ----------
+    x, y: array-like
+        1-D permutations of [1..N] vector
+    weights: array-like, optional
+        1-D array of weights. Default None equals to unit weights.
+    method: {'spearman'm 'kendalltau'}, optional
+        Defines the method to find distance:
+        'spearman' - sum of absolute distance between same elements
+        in x and y.
+        'kendalltau' - number of inverions needed to bring x to y.
+        Default is 'spearman'.
+
+    Return
+    ------
+    distance: float
+        Distance between x and y.
+
+    Example
+    -------
+    >>> from scipy.stats import rank_distance
+    >>> rank_distance([1,3,4,2],[2,3,1,4])
+    6.0
+    >>> rank_distance([1,3,4,2],[2,3,1,4], method='kendalltau')
+    4.0
+    """
+    x = np.asarray(x)
+    y = np.asarray(y)
+
+    if np.unique(x).size != x.size or np.unique(y).size != y.size:
+        raise ValueError("x and y must have only unique elements")
+
+    if x.size != y.size:
+        raise ValueError("x and y have different size")
+
+    if weights is None:
+        weights = np.ones(x.size - 1)
+    else:
+        weights = np.asarray(weights)
+        if weights.size < (x.size - 1):
```

Comment:
Two questions here:

1. What if `weights.size` is larger than `x.size`?
2. Should it be obvious why `x.size - 1` instead of just `x.size` is used? Maybe something simple that I'm missing while reviewing late at night